### PR TITLE
Infra 561 fail hard if env vars missing

### PR DESCRIPTION
**Pull request by dfuentes**
Aviator now fails if it can't find a required environment variable, with a hopefully useful message to help debug the issue.  

Example output:

```bash
$ aviator ping
Error: required environment variable 'SOME_VAR' not set.  This environment variable is referenced as 'somevar' in the 'default' setup located in '/Users/dfuentes/go/src/github.com/Clever/aviator/examples/pingpong/server/aviator.yml'.  Please set this variable in your environment and try again.
```

### CODE REVIEW

**Comment by dfuentes** [comment]
@nathanleiby simple check for now.

File: aviator/aviator.go [hunk 1]
```diff
@@ -242,8 +245,13 @@ func (a Aviator) startService(service string, useConfig string) error {
...
 			// Special case: pull from environment, if value starts with $
 			if value != "" && ([]rune(value)[0] == '$') {
-				value = os.Getenv(strings.Split(value, "$")[1])
-				log.Tracef("Reading from os env: %s\n", value)
+				envKey := strings.Split(value, "$")[1]
+				value, err := env.Get(envKey)
+				if err != nil {
+					return fmt.Errorf("required environment variable '%s' not set.  This environment variable is referenced as '%s' in the '%s' setup located in '%s'.  Please set this variable in your environment and try again.",
+						envKey, key, useConfig, activeService.ConfigPath)
+				}
```
Comment (nathanleiby) [inline aviator/aviator.go]: Would love if we could roll up the missing env keys from all services and print them together. My concern is the pain of: "A is not set..." (Set A.) "B is not set..."

Comment (nathanleiby) [inline aviator/aviator.go]: Or even just all the keys from one service

**Comment by nathanleiby** [comment]
@dfuentes reassigned to you

File: aviator/aviator.go (hunk 1 above)
Comment (dfuentes) [inline aviator/aviator.go]: @nathanleiby 0dadb15

Example output looks like:
screen shot 2015-01-05 at 5 41 34 pm

**Comment by nathanleiby** [comment]
- worth moving missing env logic to another file (still in `package aviator`)?
- automated tests on missing env logic?

**Comment by dfuentes** [comment]
Actually I'm going to take a different approach here, the initial one is flawed (even before rolling up).  

What the current implementation does is try to make sure any published values from any dependent services resolve to actual values in the environment and complains if they don't, even if those published values are not used by anything.

Ideally, what I'd want is only values that are in the required block of a service are required to be published and valid by the dependent service.  Something else that would be nice (but probably out of scope of this change) would be to allow default values, but also give an environment override from the calling environment.

I think I need to spend a bit more time thinking about exactly how environment variables should work.

**Comment by nathanleiby** [comment]
Makes good sense. I think in practice, the current approach often showed the env vars that caused problems; however, the abstraction wasn't correct.

**Comment by dfuentes** [comment]
closing in favor of https://github.com/Clever/aviator/pull/23

**Pull request closed without merging**